core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -677,9 +701,29 @@ private List<ServiceConfigSetters> getServiceConfigSetters(
         } else {
             serviceConfigSetters = ImmutableList.copyOf(this.serviceConfigSetters);
         }
+
+        if (defaultVirtualHostBuilder.contextPath() != null) {
```

Comment:
Ditto, I propose that we add prefixes at `VirtualHostBuilder#build`.

In detail, I propose the following change so that `ServiceConfigBuilder` accepts the prefix before creating the actual `ServiceConfig`.
```
+++ b/core/src/main/java/com/linecorp/armeria/server/ServiceConfigBuilder.java
@@ -293,7 +293,8 @@ final class ServiceConfigBuilder implements ServiceConfigSetters {
                         HttpHeaders virtualHostDefaultHeaders,
                         Function<? super RoutingContext, ? extends RequestId> defaultRequestIdGenerator,
                         ServiceErrorHandler defaultServiceErrorHandler,
-                        @Nullable UnhandledExceptionsReporter unhandledExceptionsReporter) {
+                        @Nullable UnhandledExceptionsReporter unhandledExceptionsReporter,
+                        String contextPath) {
         ServiceErrorHandler errorHandler =
                 serviceErrorHandler != null ? serviceErrorHandler.orElse(defaultServiceErrorHandler)
                                             : defaultServiceErrorHandler;
@@ -303,7 +304,7 @@ final class ServiceConfigBuilder implements ServiceConfigSetters {
         }
 
         return new ServiceConfig(
-                route, mappedRoute == null ? route : mappedRoute,
+                route.withPrefix(contextPath), mappedRoute == null ? route : mappedRoute,

```

This can be passed from `VirtualHostBuilder` like the following:
```
+++ b/core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java
@@ -1338,7 +1337,7 @@ public final class VirtualHostBuilder implements TlsSetters {
                                             successFunction, requestAutoAbortDelayMillis,
                                             multipartUploadsLocation, defaultHeaders,
                                             requestIdGenerator, defaultErrorHandler,
-                                            unhandledExceptionsReporter);
+                                            unhandledExceptionsReporter, contextPath);
                 }).collect(toImmutableList());
 
         final ServiceConfig fallbackServiceConfig =
@@ -1347,7 +1346,7 @@ public final class VirtualHostBuilder implements TlsSetters {
                                accessLogWriter, blockingTaskExecutor, successFunction,
                                requestAutoAbortDelayMillis, multipartUploadsLocation,
                                defaultHeaders, requestIdGenerator,
-                               defaultErrorHandler, unhandledExceptionsReporter);
+                               defaultErrorHandler, unhandledExceptionsReporter, "/");
```

Note that `fallbackServiceConfig` should still catch all requests, so the prefix isn't applied at line 1346.